spark/src/test/scala/org/apache/spark/sql/delta/DeltaProtocolVersionSuite.scala

```diff
@@ -3571,6 +3571,150 @@ trait DeltaProtocolVersionSuiteBase extends QueryTest
+      // The provenance properties should also have been removed.
+      ictProvenanceProperties.foreach(prop =>
+        assert(!snapshot.metadata.configuration.contains(prop)))
+      validateICTRemovalMetrics(usageLogs, expectProvenanceInfoRemoval = true)
+    }
+  }
+
+  test("drop InCommitTimestamp --- only one table property") {
+    withTempDir { dir =>
+      // Dropping the ICT table feature should also remove any ICT provenance
+      // table properties even when the ICT enablement table property is not present.
+      spark.sql(
+        s"CREATE TABLE delta.`${dir.getCanonicalPath}` (id bigint) USING delta" +
+          s" TBLPROPERTIES ('${DeltaConfigs.IN_COMMIT_TIMESTAMPS_ENABLED.key}' = 'true')")
+      val deltaLog = DeltaLog.forTable(spark, dir)
+      // Remove the enablement property.
+      AlterTableUnsetPropertiesDeltaCommand(
+        DeltaTableV2(spark, deltaLog.dataPath),
+        Seq(DeltaConfigs.IN_COMMIT_TIMESTAMPS_ENABLED.key),
+        ifExists = true).run(spark)
+      // Set the IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION property.
+      AlterTableSetPropertiesDeltaCommand(
+        DeltaTableV2(spark, deltaLog.dataPath),
+        Map(DeltaConfigs.IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION.key -> "1")).run(spark)
+      val snapshot1 = deltaLog.update()
+      assert(snapshot1.protocol.writerFeatureNames.contains(InCommitTimestampTableFeature.name))
+      // Ensure that the enablement property is not set.
+      assert(!DeltaConfigs.IN_COMMIT_TIMESTAMPS_ENABLED.fromMetaData(snapshot1.metadata))
+      assert(snapshot1.metadata.configuration.contains(
+        DeltaConfigs.IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION.key))
+
+      AlterTableDropFeatureDeltaCommand(
+        DeltaTableV2(spark, deltaLog.dataPath),
+        InCommitTimestampTableFeature.name)
+        .run(spark)
+      val snapshot2 = deltaLog.update()
+      assert(!snapshot2.protocol.writerFeatureNames.contains(InCommitTimestampTableFeature.name))
+      assert(!DeltaConfigs.IN_COMMIT_TIMESTAMPS_ENABLED.fromMetaData(snapshot2.metadata))
+      assert(!snapshot2.metadata.configuration.contains(
+        DeltaConfigs.IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION.key))
+    }
+  }
+
+  test("drop InCommitTimestamp --- no table property") {
+    withTempDir { dir =>
+      spark.sql(
+        s"CREATE TABLE delta.`${dir.getCanonicalPath}` (id bigint) USING delta" +
+          s" TBLPROPERTIES ('${DeltaConfigs.IN_COMMIT_TIMESTAMPS_ENABLED.key}' = 'true')")
+      val deltaLog = DeltaLog.forTable(spark, dir)
+      // Remove the enablement property.
+      AlterTableUnsetPropertiesDeltaCommand(
+        DeltaTableV2(spark, deltaLog.dataPath),
+        Seq(DeltaConfigs.IN_COMMIT_TIMESTAMPS_ENABLED.key),
+        ifExists = true).run(spark)
+
+      AlterTableDropFeatureDeltaCommand(
+        DeltaTableV2(spark, deltaLog.dataPath),
+        InCommitTimestampTableFeature.name)
+        .run(spark)
```

Comment:
Can we also add assertion over usage logs?

Reply:
Done